src/unix/linux-core.c

```diff
@@ -1014,15 +1015,257 @@ uint64_t uv_get_total_memory(void) {
+  uint8_t cgroups_version;
+  /* Path in which param files are found for a subsystem. */
+  char path[4096];
+} uv__cgroups_subsystem_info_t;
+
+
+/*
+ * Writes `start` to `field_start` and the length of the current
+ * whitespace-delimited field to `field_len`, and returns an pointer to the
+ * remainder of the string (excluding the whitespace).
+ * Note that we assume that `start` ends with "\n\0".
+ */
+static const char* uv__mountinfo_next_field(const char* start,
+    const char** field_start, size_t* field_len) {
+  size_t len;
+  const char* field_end;
+  if ('\0' == *start)
+    return 0;
+  field_end = strchr(start, ' ');
+  if (field_end == NULL) {
+    len = strlen(start) - 1; /* -1 to lop off newline */
+  } else {
+    len = field_end - start;
+  }
+  if (field_start != NULL) {
+    *field_start = start;
+  }
+  if (field_len != NULL) {
+    *field_len = len;
+  }
+  return start + len + 1;
+}
+
+
+/*
+ * Given a comma-delimited sequence of strings `super_options` (w/ size
+ * `super_options_len`), return TRUE if option exactly matches one of the
+ * elements in that sequence.
+ */
+static int uv__mountinfo_find_super_option(const char* super_options,
+    size_t super_options_len, const char* option) {
+  size_t option_len = strlen(option);
+  const char* low = super_options;
+  const char* high = super_options + super_options_len - option_len;
+  const char* preceding_char;
+  const char* succeeding_char;
+  while (strlen(low)) {
+    low = strstr(low, option);
+    if (low == NULL || low > high)
+      break;
+    preceding_char = low - 1;
+    succeeding_char = low + option_len;
+    if (low == super_options || ',' == *preceding_char) {
+      /*
+       * The occurrence of `option` is preceded by a comma, or is part of the
+       * first entry in `super_options`.
+       */
+      if (' ' == *succeeding_char || ',' == *succeeding_char || '\n' == *succeeding_char)
+        /*
+         * The occurrence of `option` is succeeded by a comma, or is part of the
+         * last entry in `super_options`.
+         */
+        return 1;
+    }
+  }
+  return 0;
+}
+
+
+/*
+ * Read /proc/self/mountinfo and /proc/self/cgroup to get info about how the
+ * given subsystem is controlled for this process, and the path to the
+ * subsystem's parameter files, if possible.
+ */
+static int uv__read_cgroups_proc_files(uv__cgroups_subsystem_info_t* info, const char* subsystem) {
+  FILE* fp;
+  /* Should be big enough to fit a single line. */
+  char buf[4096];
+  /*
+   * See description of /proc/[pid]/mountinfo in proc(5).
+   * Henceforth, numbers in parentheses refer to fields from the docs.
+   */
+  const char* field_ptr;
+  const char* field_start[10];
+  size_t field_len[10];
+  size_t field_idx;
+  char root[1024]; /* (4) */
+  char mount_point[1024]; /* (5) */
+  /* From /proc/self/cgroup */
+  char hierarchy_path[1024];
+  /* Should be big enough to fit a subsystem name, plus 2 more characters. */
+  char subsystem_search_string[64];
```

Comment:
Is it documented somewhere that subsystems will be no more than 62 bytes?

Reply:
It should actually be no more than 10, the longest name is "perf_event". I don't think there are arbitrary subsystems (at least any with parameter files) so that's as long as it should get. I've reduced this down to `16`.

Reply:
Clean code is obviously correct. Instead of relying on implementation details, I think you should write this in a way that doesn't depending on deep knowledge and point-in-time implementation details. Code should be obviously correct to a casual reader.

My suggestion is to allocate this dynamically based on the length of the subsystem string passed in instead of relying on constants.